server/py/framework/db/sqldb/db.py

```diff
@@ -5887,6 +5888,83 @@ def _delete_alert_notifications(
         if commit:
             session.commit()
 
+    def store_alert_activation(
+        self,
+        session,
+        alert_data: mlrun.common.schemas.AlertConfig,
+        event_data: mlrun.common.schemas.Event,
+        notifications_states: list[mlrun.common.schemas.NotificationState],
+    ):
+        extra_data = {
+            "notifications": [
+                notification.dict() for notification in notifications_states
+            ],
+            "criteria": alert_data.criteria.dict() if alert_data.criteria else None,
+        }
+
+        # For JOB entities, construct entity_id as "name.uid" format
+        if alert_data.entities.kind == mlrun.common.schemas.alert.EventEntityKind.JOB:
+            run_name = alert_data.entities.ids[0]
+            run_uid = event_data.value_dict.get("uid")
+            entity_id = f"{run_name}.{run_uid}"
+        else:
+            entity_id = alert_data.entities.ids[0]
+
+        alert_activation_record = AlertActivation(
+            name=alert_data.name,
+            project=alert_data.project,
+            activation_time=event_data.timestamp,
+            entity_id=entity_id,
+            entity_kind=alert_data.entities.kind,
+            event_kind=event_data.kind,
+            severity=alert_data.severity,
+            number_of_events=alert_data.criteria.count,
+            data=extra_data,
+        )
+
+        self._upsert(session, [alert_activation_record])
+
+    def list_alert_activations(
+        self, session: Session, project: typing.Optional[str] = None
+    ) -> list[mlrun.common.schemas.AlertActivation]:
+        # TODO: add filters
+        query = self._query(session, AlertActivation)
+
+        if project and project != "*":
+            query = query.filter(AlertActivation.project == project)
+
+        alert_activations = list(
+            map(self._transform_alert_activation_record_to_scheme, query.all())
+        )
+
+        return alert_activations
```

Comment:
This is more readable and pythonic than using map
```suggestion
        return [
            self._transform_alert_activation_record_to_scheme(record)
            for record in query.all()
        ]
```